docs/client-concepts/high-level/indexing/indexing.asciidoc

```diff
@@ -0,0 +1,186 @@
+{
+    <3>
+    foreach (var itemWithError in indexManyResponse.ItemsWithErrors)
+    {
+        Console.WriteLine("Failed to index document {0}: {1}", itemWithError.Id, itemWithError.Error);
+    }
+}
+
+// Alternatively, documents can be indexed asynchronously
+var indexManyAsyncResponse = await client.IndexManyAsync(people); <4>
+----
+<1> synchronous method that returns an `IBulkResponse`
+
+<2> the response can be inspected for its success
+
+<3> If there are errors, they can be enumerated and inspected
+
+<4> asynchronous method that returns a `Task<IBulkResponse>` that can be awaited
+
+==== Multiple Documents with Bulk
+
+If you require finer grained control over bulk indexing you can use the `Bulk` and `BulkAsync` methods and use the descriptors to
+customise the bulk calls.
+
+[source,csharp]
+----
+var bulkIndexResponse = client.Bulk(b => b
+    .Index("people")
+    .IndexMany(people)); <1>
+
+// Alternatively, documents can be indexed asynchronously similar to `IndexManyAsync`
+var asyncBulkIndexResponse = await client.BulkAsync(b => b
+    .Index("people")
+    .IndexMany(people)); <4>
+----
+<1> synchronous method that returns an `IBulkResponse`, the same as `IndexMany` and can be inspected in the same way for errors
+
+<4> asynchronous method that returns a `Task<IBulkResponse>` that can be awaited
+
+==== Multiple Documents with BulkAllObservable helper
+
+Multiple documents can be indexed using the `BulkAllObservable` helper. This helper exposes retry and backoff functionality
+to automatically retry in the event of a failure. This allows you to focus on the overall objective of indexing,
+without having to concern yourself with retry mechanics. Of course, if there is an eventual failure you have
+the ability to inspect any exceptions and indexing progress.
+
+[source,csharp]
+----
+var people = new []
+{
+    new Person
+    {
+        Id = 1,
+        FirstName = "Martijn",
+        LastName = "Laarman"
+    },
+    new Person
+    {
+        Id = 2,
+        FirstName = "Stuart",
+        LastName = "Cam"
+    },
+    new Person
+    {
+        Id = 3,
+        FirstName = "Russell",
+        LastName = "Cam"
+    }
+};
+
+var bulkAllObservable = client.BulkAll(people, b => b
+    .Index("people")
+    .BackOffTime("30s") <1>
+    .BackOffRetries(2) <2>
+    .RefreshOnCompleted()
+    .MaxDegreeOfParallelism(Environment.ProcessorCount)
+    .Size(1000) <3>
+);
+
+Exception exception = null;
+var waitHandle = new ManualResetEvent(false);
+
+bulkAllObservable.Subscribe(new BulkAllObserver( <4>
+    onNext: b =>
+    {
+        // Do something e.g. write number of pages to console
+    },
+    onError: e =>
+    {
+        exception = e; <5>
+        waitHandle.Set();
+    },
+    onCompleted: () => waitHandle.Set()));
+
+waitHandle.WaitOne(); <6>
+
+if (exception != null) <7>
+    throw exception;
+----
+<1> how long to wait between retries
+
+<2> how many reties should this bulk index attempt is unsuccessful
```

Comment:
nitpick: retries

I think this explanation could be elaborated on a little.

Reply:
++